proposals/872-spark-history-server-mcp/README.md

```diff
@@ -0,0 +1,341 @@
+
+The Spark History Server is essential for post-mortem analysis and performance optimization of Spark workloads, but currently requires manual navigation through complex web interfaces to extract insights.
+
+## Motivation
+
+Kubeflow users running Spark workloads currently lack AI-powered troubleshooting capabilities. While the Spark Operator handles job lifecycle management, users must manually analyze Spark UI logs, metrics, and performance data when jobs fail or perform poorly.
+
+### Goals
+
+- **Enable Natural Language Spark Analysis**: Provide conversational AI interface for Spark troubleshooting and performance analysis
+- **Integrate with Kubeflow Ecosystem**: Seamlessly connect with existing Spark Operator and History Server infrastructure
+- **AI-Powered Insights**: Deliver intelligent recommendations for Spark job optimization and failure diagnosis
+- **Standardized AI Integration**: Implement Model Context Protocol (MCP) standard for consistent AI agent interactions
+- **Multi-Framework Support**: Enable compatibility with various AI tools (Claude, Amazon Q, LangGraph, etc.)
+
+### Non-Goals
+
+- **Replace Existing Monitoring**: Not intended to replace Spark UI, Grafana, or other established monitoring tools
+- **Real-time Job Monitoring**: Focus only on completed job analysis, not live job monitoring
+- **Modify Core Spark Components**: No changes to Spark History Server core functionality or Spark Operator
+- **Custom Spark Distribution**: Not creating a fork or custom version of Apache Spark
+- **Direct Database Access**: Will not bypass Spark History Server APIs or access event logs directly
+
+### Problem Statement
+- Manual analysis of Spark performance issues is time-consuming
+- Lack of intelligent troubleshooting capabilities  
+- No natural language interface for Spark observability
+- Disconnected monitoring tools that don't integrate with AI workflows
+- Complex Spark History Server web UI requires expertise to navigate effectively
+
+**Related GitHub Issue**: https://github.com/kubeflow/community/issues/872
+
+## Architecture Overview
+
+```mermaid
+graph TB
+    subgraph "User Interface"
+        USER[Users via Web UI<br/>Amazon Q CLI, Claude, LangGraph, Strands]
+    end
+
+    subgraph "AI Analysis Layer"
+        MCP[Spark History Server MCP<br/>AI-Powered Analysis<br/>📋 THIS PROPOSAL]
+    end
+
+    subgraph "Spark Infrastructure"
+        SHS[Spark History Server<br/>Event Logs & Metrics]
+        SO[Spark Operator<br/>Running Spark Jobs]
+    end
+
+    USER -->|Natural Language Queries| MCP
+    MCP -->|Fetch Data via REST API| SHS
+    SO -->|Writes Event Logs| SHS
+    MCP -->|AI Analysis & Recommendations| USER
+
+    classDef user fill:#e1f5fe
+    classDef proposal fill:#ffeb3b,stroke:#f57f17,stroke-width:3px
+    classDef spark fill:#fff3e0
+
+    class USER user
+    class MCP proposal
+    class SHS,SO spark
+```
+
+
+## Kubeflow Checklist
```

Comment:
@vara-bonthu Please can you also add CNCF Checklist and Background information according to this Application template: https://github.com/kubeflow/community/blob/master/how-to/join_kubeflow_ecosystem.md#cncf-short-checklist ?